SpreadCheetah/Styling/NumberFormat.cs

```diff
@@ -0,0 +1,93 @@
+using SpreadCheetah.Helpers;
+using System.Globalization;
+
+namespace SpreadCheetah.Styling
+{
+    /// <summary>
+    /// Format that defines how a number or <see cref="DateTime"/> cell should be displayed.
+    /// May be a custom format, initialised by <see cref="NumberFormat.Custom"/>, or a standard format, initialised by <see cref="NumberFormat.Standard"/>
+    /// </summary>
+#pragma warning disable CA1815, CA2231 // Override equals and operator equals on value types - equals operator not appropriate for this type
+    public readonly struct NumberFormat : IEquatable<NumberFormat>
+#pragma warning restore CA1815, CA2231
+    {
+        private NumberFormat(string? customFormat, int? standardFormat)
+        {
+            CustomFormat = customFormat.WithEnsuredMaxLength(255);
+            StandardFormat = standardFormat;
+        }
+        
+        internal string? CustomFormat { get; }
+        internal int? StandardFormat { get; }
```

Comment:
If we use `StandardNumberFormat?` as the type instead of `int?` then we can simplify the `ToString()` implementation below.

Reply:
My initial thinking was that we might want some way of representing format numbers that are not in the enum, but I guess that's something that could be addressed in the future if it ever is required.